contracts/EverlongStrategy.sol

```diff
@@ -0,0 +1,478 @@
+//    ####  ###########++++++--+++++##++++++++++++++##+++++++++++####             #
+//  +#################+++++--+++++++###+#++++++++++++####++++++++#######
+// +##################++++++++++++++++++##+++++++++++###+++++++++#########
+// ##################++++++++++++++++++++++++++++++++++###+++++++###############
+// ##################++-+++++++++++++++++++++++++++++++##+++++++###################
+//   ###    #####++++++-++++++++++++++++++++++++++++++++++++++++##########  #######
+//  ###      ###+++++++++---+++++++++++++++-+++-++++++++++++++++######### #########
+//          ###++++++++------++++++++++++###++-++++++++--+++++++#############
+//        #####+++++++----------+++++++++++++++##++++-------+++##########
+//    #########++++++----------+++++++++++++++++++++++------+++#########
+// #############+++++----------+++++++++++++++++++++++++---+++###########
+// #############++-++----------+++++++++++++++++++++++-+---++###    ## ###
+// ### ###       ++++----------+++++++-++-++++++-+++++-----++###        ###
+//                 ++----------+++#+++---------+++--++-+---++#####       ####
+//                #+----------+++++++++++++--+--+++-------+-++#####       ####
+//              ####+---------++++++++++++++++++#+++-----++-++#######      ####
+//            ######+----------------++++++++++++++-----++--+    #####       ###
+//          ##### #+++------+++-------+++++++--+++-----++-###       ###      #####
+//         ###    ++--+-----+++++++++++++---------+---++-##++###    +##
+//        ##+   ###+--------+++++++++++++++++--+++---++++      ++#######
+//     ####+    ###+-----------+++++++++++++++++++++++++            #  #
+//           ++##++++---------------+++++++++++++++++
+//          ++## +++++------------+++++++++++++++++++
+//              +++++++-----------+++++++++++++++++++
+//          ++###++++++---------------++++++++++++++
+//       +++#####+++++++++--------------++++++++++++
+// #+ +++#######+++++++++++---------------+++++++++#+
+// ###########++++++++++++++-------------++++++++++###
+//
+/// @author DELV
+/// @title EverlongStrategy
+/// @notice A money market powered by Hyperdrive.
+/// @custom:disclaimer The language used in this code is for coding convenience
+///                    only, and is not intended to, and does not, have any
+///                    particular legal or regulatory significance.
+contract EverlongStrategy is BaseStrategy {
+    using FixedPointMath for uint256;
+    using HyperdriveExecutionLibrary for IHyperdrive;
+    using Portfolio for Portfolio.State;
+    using SafeCast for *;
+    using SafeERC20 for ERC20;
+    using HyperdriveUtils for *;
+
+    /// @notice Amount of additional bonds to close during a partial position
+    ///         closure to avoid rounding errors. Represented as a percentage
+    ///         of the positions total  amount of bonds where 1e18 represents
+    ///         a 10% buffer.
+    uint256 public constant partialPositionClosureBuffer = 0.001e18;
```

Comment:
Note: I'll have to come back to this but representing 10% as 1e18 seems confusing. We can't we just have 1e18 represent 100% like normal?

Reply:
typo 🙃